### ISSUE TITLE
[Wiki] State Machine explanations

### ISSUE THREAD

**Issue by XkyRauh** [malware-dev/MDK-SE#72]
Hi!  

Currently the State Machine utilizing `yield` establishes a good State Machine that is entirely contained within the Main() method of the PB.  

Multiple people were asking on the #programming-in-game channel of the KeenSWH Discord how to incorporate yield into multiple methods.  

Digi was kind enough to post a snippet of pseudocode that went like this:

```cs
// in your statemachine thing
foreach(var thing in DoStuff())
{
    yield return thing;
}

// the method
IEnumerable<WhateverType> DoStuff()
{
    //...
    yield return ...;
    // etc
}
```
It has been useful enough that I think it deserves a place in the Wiki!

**Comment by malware-dev** [comment]
Good idea. I will write something up when I can.

**Comment by XkyRauh** [comment]
That's because I was still learning how this stuff worked when I encountered the error.  :)  This suggested edit is unnecessary, because at this point I thought I'd need a separate Enumerable for each State Machine, instead of re-using the same one over and over again.

As per the lessons learned from KiwiNinja, I now understand to set the State Machine to whatever IEnumerable<Type> I'd like to step through, and then set the UpdateFrequency |= Once to activate the State Machine. 

THAT is what should be edited into the Wiki, not the trash I suggested above!!
